src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala

```diff
@@ -0,0 +1,194 @@
+package com.microsoft.partnercatalyst.fortis.spark
+
+import com.microsoft.partnercatalyst.fortis.spark.logging.AppInsights
+import com.microsoft.partnercatalyst.fortis.spark.pipeline._
+import com.microsoft.partnercatalyst.fortis.spark.streamprovider.ConnectorConfig
+import com.microsoft.partnercatalyst.fortis.spark.transforms.image.{ImageAnalysisAuth, ImageAnalyzer}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.language.{LanguageDetector, LanguageDetectorAuth}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.{Geofence, LocationsExtractor, PlaceRecognizer}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.client.FeatureServiceClient
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.{SentimentDetector, SentimentDetectorAuth}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.topic.KeywordExtractor
+import org.apache.log4j.{Level, LogManager, Logger}
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.streaming.{Seconds, StreamingContext}
+import sun.reflect.generics.reflectiveObjects.NotImplementedException
+
+object ProjectFortis extends App {
+
+  private val pipelines = List[Pipeline](
+    InstagramPipeline,
+    RadioPipeline,
+    TwitterPipeline,
+    FacebookPipeline,
+    TadawebPipeline
+  )
+
+  // TODO: create helper method to augment Settings object with data from Cassandra
+  // TODO: move most of these settings to Cassandra
+  private implicit object Settings extends Settings {
+    import scala.util.Properties.envOrNone
+
+    val progressDir: String = envOrFail(Constants.Env.HighlyAvailableProgressDir)
+    val featureServiceHost: String = envOrFail(Constants.Env.FeatureServiceHost)
+    val oxfordLanguageToken: String = envOrFail(Constants.Env.OxfordLanguageToken)
+    val oxfordVisionToken: String = envOrFail(Constants.Env.OxfordVisionToken)
+
+    val appInsightsKey: Option[String] = envOrNone(Constants.Env.AppInsightsKey)
+    val modelsDir: Option[String] = envOrNone(Constants.Env.LanguageModelDir)
+
+    private def envOrFail(name: String): String = {
```

Comment:
I'd move this out to a utility as the function implementation is unrelated to the code at hand and so just takes up space when trying to scan the code. The brain's L1 cache is pretty limited so we should keep code at the [same level of abstraction](https://dzone.com/articles/the-single-level-of-abstraction-principle).